src/charm.py

```diff
@@ -366,7 +376,10 @@ def _common_exit_hook(self) -> None:
             service := self.container.get_service(self._service_name)
         ) and service.is_running()
 
-        num_peers = len(self.peer_relation.units)
+        num_peers = 0
+
+        if self.peer_relation:
+            num_peers = len(self.peer_relation.units)
```

Comment:
Might be better if we call `get_relation` less times.
If you're in the mood for mini-golfing:

```suggestion
        num_peers = len(rel.units) if (rel := self.peer_relation) else 0
```

I forget - does rel.units include the querying unit? I.e. should we also have:
```python
num_peers = max(0, num_peers - 1)
```

Reply:
No, it does not.